src/v/coproc/offset_storage_utils.cc

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2020 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "coproc/offset_storage_utils.h"
+
+#include "bytes/iobuf.h"
+#include "coproc/logger.h"
+#include "model/fundamental.h"
+#include "reflection/adl.h"
+#include "vlog.h"
+
+#include <seastar/core/coroutine.hh>
+
+#include <optional>
+
+namespace coproc {
+
+void offsets_map_to_iobuf(
+  iobuf& buf, const ntp_context::offset_tracker& offsets) {
+    reflection::serialize(buf, offsets.size());
+    for (const auto& e : offsets) {
+        reflection::serialize(
+          buf, e.first, e.second.last_read, e.second.last_acked);
+    }
+}
+
+ntp_context::offset_tracker rebuild_offsets_map(iobuf_parser& p) {
+    ntp_context::offset_tracker offsets;
+    auto size = reflection::adl<ntp_context::offset_tracker::size_type>{}.from(
```

Comment:
use a fixed size data type (like int32_t) that is independent of the interface / data structure choice.